### PR TITLE
Completed week 2 assessment

### PR DESCRIPTION
**Pull request by patsysaurer**
# Submitting Week 2 Assessment

### ✅ Completed Files

Select the following as confirmation that the files in this repository are complete.

- [✅ ] Career Readiness
- [✅] Coding Challenges
- [✅ ] Coding Critique
- [✅ ] Coding Concepts
- [✅ ] Interview Questions

### ✋ Reminder

Do NOT merge your assessment!

### 📝 Feedback

If you would like specific feedback please leave details here:

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -25,27 +27,71 @@ const numbersArray2 = [24, 27, 30, 33, 36]
...
+        const object1 = { number: 15 }
+        // Expected output: "15 is divisible by three"
+        const object2 = { number: 0 }
+        // Expected output: "0 is divisible by three"
+        const object3 = { number: -7 }
+        // Expected output: "-7 is not divisible by three"  
+
+      expect(divisibleByThree(object1.number)).toEqual("15 is divisible by three")
+      expect(divisibleByThree(object2.number)).toEqual("0 is divisible by three")
+      expect(divisibleByThree(object3.number)).toEqual("-7 is not divisible by three")
```
Comment (chrisyhlee) [inline code-challenges.test.js:42]: The prompt states the function should take an object as an argument. Here, you are passing a number as the argument.

File: code-challenges.test.js [hunk 2]
```diff
@@ -25,27 +27,71 @@ const numbersArray2 = [24, 27, 30, 33, 36]
...
+// Pseudo code:
+// Create a function named capitalizedWords
+// Input: array
+// Output: array with all words capitalized
+const capitalizedWords = (array) => {
+    // iterate using .map() 
+    return array.map(word => {
+        // each iteration use the .toUppercase()
+        
+       return word[0].toUppercase()
```
Comment (chrisyhlee) [inline code-challenges.test.js:93]: This is currently returning the first letter of each word capitalized. What you need is to also add the rest of the word!

File: code-challenges.test.js [hunk 3]
```diff
@@ -25,27 +27,71 @@ const numbersArray2 = [24, 27, 30, 33, 36]
...
+    })
+}
+
+// I need more practice with Jest!
```
Comment (chrisyhlee) [inline code-challenges.test.js:97]: Your jest test is picture perfect, you just missed a step in your code, which I commented on above

File: code-critique.js [hunk 4]
```diff
@@ -6,5 +6,44 @@
...
+// const firstVowel = (string) => {
+//     const vowels = "AEIOUaeiou"
+//     let firstVowel = []
+//     let stringArr = string.split("")
+//     stringArr.filter(value, index => index === )
+//     return firstVowel.join("")
+// } 
+// console.log(firstVowel("Patsy"))
+
+// I learned that there are many ways to solving one problem. Also many languages that are out there. The format and most keywords seem similar to JavaScript so that makes it easier to get an idea of what is going on.
```
Comment (chrisyhlee) [inline code-critique.js:49]: Good takeaway. Some other things to consider:
- Would this be helpful to you if this was a question you had for the AI?
- Is everything in the answer actually useful information, or is there unnecessary info in there?
- Could you change the prompt to get a better answer?

File: coding-concepts.js [hunk 5]
```diff
@@ -9,28 +9,28 @@
 // --------------------1) What will this log?
 
 const cohort = "Golf 2023"
-// console.log(cohort.split(" "))
+console.log(cohort.split(" "))
 
-// a) Your answer:
-// b) Verify and explain:
+// a) Your answer: ["G","o","l","f"," ","2","0","2","3"]
+// b) Verify and explain: Wrong. The space in between the quotes in the .split() allowed the words to be in their own index.
```
Comment (chrisyhlee) [inline coding-concepts.js:15]: Great explanation

File: coding-concepts.js [hunk 6]
```diff
@@ -55,7 +55,7 @@ class Learn {
   }
 }
 const learnStudent = new Learn("George")
-// console.log(learnStudent)
+console.log(learnStudent)
 
-// a) Your answer:
-// b) Verify and explain:
+// a) Your answer: I don't think it will log anything. I think it is missing the dot notation to access the student. 
+// b) Verify and explain: Wrong. It totally logged the entire object with "George" as the student. Interesting. I need more practice :)
```
Comment (chrisyhlee) [inline coding-concepts.js:61]: Practice is key!

**Review by chrisyhlee — APPROVED** [review]
Another great assessment! This shows Advanced Beginner. You have a great understanding of the logic, and I'm a big fan of the line by line pseudo code. I like that you're keeping the pseudo code short but sweet, explaining what that line of code is doing! I've left some comments you can scroll through, I hope they help you. My suggestion for you moving forward is to continue practicing and getting the reps in every day. You're doing awesome!

File: code-challenges.test.js (hunk 1 above)
Comment (patsysaurer) [reply code-challenges.test.js:42]: Got it thanks!

File: coding-concepts.js (hunk 6 above)
Comment (patsysaurer) [reply coding-concepts.js:61]: Thank you Chris!

**Pull request merged**